File: Back/Users/src/DB/models/order.js
import mongoose from "mongoose";
// import Joi from "joi";
// import joiObjectid from "joi-objectid";
// Joi.objectId = joiObjectid(Joi);

const orderSchema  = new mongoose.Schema(
    {
        product : {
            type: {
                productID : {type : mongoose.Schema.Types.ObjectId , ref: "products" , required: true },
                price : {type : Number ,required: true},
                color : {type : String ,required: true},
                garantee : {type : Number ,required: true},
                sellerID : {type : mongoose.Schema.Types.ObjectId , ref: "products" , required: true },
                picture : {type : String ,required: true}
            },
            required : true},
        quantity: {type : Number ,required: true , default: 1},
        userID : {type : mongoose.Schema.Types.ObjectId , ref: "products" , required: true },
        rate : {type : Number ,required: true , default: 1},
        state : {type: String, enum: ["delivered" , "pending" , "canceled" , "returned" , "preparingOrder" , "transit" , "notOrdered"], required: true},
        usedDate : {type: Date, required : true, },  
    }
);

export const OrderModel = mongoose.model("orders",orderSchema);

File: Back:Products/models/order.go
package models

import (
	"time"

	"go.mongodb.org/mongo-driver/bson/primitive"
)

type order_product struct {
	ProdID    primitive.ObjectID
	Price     int
	Color     string
	Guarantee Guarantee
	SellerID  primitive.ObjectID
	Picture   string
}

type StateType int

const (
	Delivered StateType = iota + 1
	Pending
	Canceled
	Returned
	PreparingOrder
	Transit
	NotOrdered
)

func (d StateType) String() string {
	return [...]string{"delivered", "pending", "canceled", "returned", "preparingOrder", "transit", "notOrdered"}[d-1]
}

func (d StateType) EnumIndex() int {
	return int(d)
}

type Order struct {
	ID          primitive.ObjectID `json:"_id,omitempty" bson:"_id,omitempty"`
	Product     order_product      `json:"product" bson:"product"`
	Quantity    int                `json:"quantity" bson:"quantity"`
	UserID      primitive.ObjectID `json:"user_id" bson:"user_id"`
	Rate        int                `json:"rate,omitempty" bson:"rate,omitempty"`
	State       StateType          `json:"state" bson:"state"`
	ReceiveDate time.Time          `json:"receive_date,omitempty" bson:"receive_date,omitempty"`
}
